File: python_implementation/src/application/use_cases/tasks/get_task_by_id_use_case.py
from typing import Optional
from uuid import UUID
from src.domain.abstractions.repositories.abstract_task_repository import AbstractTaskRepository
from src.domain.entities.task_entity import TaskEntity


class GetTaskByIdUseCase:
    """Use case for retrieving a task by ID"""
    
    def __init__(self, task_repository: AbstractTaskRepository):
        self.task_repository = task_repository
    
    def execute(self, task_id: UUID) -> Optional[TaskEntity]:
        return self.task_repository.get_by_id(task_id)

File: java_implementation/src/main/java/com/example/taskapi/application/usecases/GetTaskByIdUseCase.java
package com.example.taskapi.application.usecases;

import com.example.taskapi.domain.model.Task;
import com.example.taskapi.domain.repository.TaskRepositoryPort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class GetTaskByIdUseCase {
    private final TaskRepositoryPort taskRepository;

    public Optional<Task> execute(UUID taskId) {
        return taskRepository.getById(taskId);
    }
}
